src/test/java/com/squareup/javapoet/SimpleElementName.java

```diff
@@ -0,0 +1,39 @@
+package com.squareup.javapoet;
+
+import javax.lang.model.element.Name;
+
+/**
+ * Simple implementation of {@link Name} for testing.
+ */
+public class SimpleElementName implements Name {
```

Comment:
off topic: it’s so weird that this API _is a _ CharSequence rather than _has a_ char sequence.